tensorflow/python/distribute/input_lib.py

```diff
@@ -1602,7 +1605,7 @@ def _make_iterator(self):
     if not self._worker:
       raise ValueError("Worked device must be specified when creating an "
                        "owned iterator.")
-    if self._replication_mode == InputReplicationMode.PER_WORKER:
+    if self._options is None or not self._options.experimental_copy_dataset_on_device :
```

Comment:
I think there are a few cases to consider here:

replication_mode == PER_WORKER => keep the current code 


replication_mode == PER_REPLICA =>
   experimental_prefetch_to_device == True => use `OwnedMultiDeviceIterator`
   experimental_prefetch_to_device == False => use `iter(..)`


Do you need to place the iterator on the GPU as well? I thought in the morning you had mentioned that you only need to place the dataset and not the iterator. But if you do need that, then yeah there is another branch which does 

experimental_copy_dataset_on_device == False => "with ops.device(self._worker):" 
experimental_copy_dataset_on_device == True => "with ops.device(self._devices[0]):"

Reply:
logic added per discussion.